oak/server/rust/oak_runtime/src/node/grpc/client.rs

```diff
@@ -0,0 +1,258 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use http::uri::Uri;
+use log::{debug, error, info, warn};
+use std::{
+    fmt::{self, Display, Formatter},
+    thread::{self, JoinHandle},
+    time::Duration,
+};
+use tonic::transport::{Certificate, Channel, ClientTlsConfig};
+
+use oak_abi::{
+    proto::oak::encap::{GrpcRequest, GrpcResponse},
+    OakStatus,
+};
+
+use crate::{
+    node::{
+        grpc::{
+            codec::VecCodec,
+            io::{Invocation, Receiver},
+        },
+        Node,
+    },
+    pretty_name_for_thread,
+    runtime::RuntimeProxy,
+    Handle,
+};
+
+/// Struct that represents a gRPC client pseudo-Node.
+///
+/// Reads gRPC invocations from the [`GrpcClientNode::receiver`], sends gRPC requests to an external
+/// gRPC service and writes gRPC responses back to the invocation channel.
+pub struct GrpcClientNode {
+    /// Pseudo-Node name that corresponds to an entry from the [`ApplicationConfiguration`].
+    ///
+    /// [`ApplicationConfiguration`]: crate::proto::oak::application::ApplicationConfiguration
+    config_name: String,
+    /// Reference to a Runtime that corresponds to a Node that created a gRPC client pseudo-Node.
```

Comment:
I think the `RuntimeProxy` refers to the pseudo-Node itself, not the Node that created it.

Reply:
Indeed.